README tweaks for 2.0.0 (#309)
webrecorder/replayweb.page: README.md

> [Embedded Viewer](https://replayweb.page/docs/embedding/) Usage Stats: [![](https://data.jsdelivr.com/v1/package/npm/replaywebpage/badge)](https://www.jsdelivr.com/package/npm/replaywebpage)

Reviewer: Would revert moving this to the top, the h1 should generally be the first thing in the document.

> ## Serverless Web Archive Replay

Reviewer: Replay is in the name already, maybe give new folks some more adjectives to think about when understanding what the app does?  Alternatively, this heading could just be removed.

```suggestion
## Serverless Web Archive Viewer
```

> ReplayWeb.page provides a full web archive replay system running directly in the browser,

Reviewer: If the heading is getting removed, would change here instead.

```suggestion
ReplayWeb.page provides a full web archive viewer that runs directly in the browser,
```

available at: [https://replayweb.page/](https://replayweb.page)

For full user docs, see: [https://replayweb.page/docs](https://replayweb.page/docs).

The ReplayWeb.page App can be downloaded from the [Releases](https://replayweb.page/releases) page.

See [CHANGES.md](CHANGES.md) for the current changes, or the release notes on the link above.

### Embedding Guide

See the [Embedding Guide](https://replayweb.page/docs/embedding/) for more info on embedding web archives in other sites.

## What's in this repo

> ReplayWeb.page provides a static site generated with mkdocs, an npm package/library, and an Electron app all in this repo.

Reviewer: ```suggestion
ReplayWeb.page provides a static site generated with MkDocs, an npm package/library, and an Electron app all in this repo.
```

This repository contains the 'frontend' UI for the replay system, while the 'backend' is provided via a service worker
implementation found at: https://github.com/webrecorder/wabac.js. (Of course, both frontend and backend actually run in the browser).

The frontend is loaded from `ui.js`, while the backend service/web worker is loaded from `sw.js`.

This repository contains:

- The core site hosted on https://replayweb.page/ via GitHub Pages

> - The docs for https://replayweb.page/docs` created with mkdocs

Reviewer: ```suggestion
- The docs for https://replayweb.page/docs created with [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/)
```

- The package for npm module: https://www.npmjs.com/package/replaywebpage
- A build system for https://replayweb.page and ReplayWeb.page App.
- App releases at: https://github.com/webrecorder/replayweb.page/releases

## Running ReplayWeb.page

To run ReplayWeb.page and view web archives, a regular HTTP server is all that is needed.

> ReplayWeb.page can run with any HTTP server locally. For example, you can run `http-server -p 9990` or `python -m http.server 9990` to run a static web server in the directory of this repository after running `yarn run build`. Then, simply load `http://localhost:9990/` and you'll have the core replayweb.page running locally.

Reviewer: ```suggestion
ReplayWeb.page can run with any HTTP server locally. For example, you can run `http-server -p 9990` or `python -m http.server 9990` to run a static web server in the directory of this repository after running `yarn run build`. Then, load `http://localhost:9990/` and you'll have the core replayweb.page running locally.
```

## Developing ReplayWeb.page

ReplayWeb.page is built as a Node package can be installed using yarn: `yarn install`

The package provides various commands that can be used with yarn/npm. Some useful commands include:

- `yarn start-dev` - to run in dev mode with Webpack dev server on port 9990. Autobuilds dev assets.

- `yarn start-docs` - to build assets in dev mode, and start docs in dev mode. Autobuilds /docs assets.

- `yarn start-prod` - to run production site on port 9990, with previously built assets (without docs)

- `yarn start-prod-docs` - to build production version of the site and build docs with `/docs` endpoint available.

- `yarn start-electron` - to start electron in dev mode, with previously built assets

- `yarn build` - to build production assets sw.js, ui.js

- `yarn build-docs` - to build the docs site in `mkdocs/site` to `mkdocs/_genhtml`

- `yarn dist` - to build production assets + Electron app (in dist/)

### Static Site

> The static assets are placed in the root `index.html`, `sw.js` and `ui.js`, and can be used with any HTTP server. This provides the core ReplayWeb.page functionality.

Reviewer: ```suggestion
The static assets are placed in the root `index.html`, `sw.js`, and `ui.js`, and can be used with any HTTP server. This provides the core ReplayWeb.page functionality.
```

### Static Site + Docs

> The full site with docs is built using mkdocs. Using the `yarn build-docs` command, the static assets are copied into `mkdocs/site/` and using markdown in `mkdocs/site/docs`, the final static site is built to `mkdocs/_genhtml`. This is what is published to https://replayweb.page/ via CI.

Reviewer: ```suggestion
The full site with docs is built using MkDocs. Using the `yarn build-docs` command, the static assets are copied into `mkdocs/site/` and using Markdown in `mkdocs/site/docs`, the final static site is built to `mkdocs/_genhtml`. This is what is published to https://replayweb.page/ via CI.
```
